migrations/schemas/20221011152829-quests.sql

```diff
@@ -0,0 +1,109 @@
+
+-- +migrate Up
+CREATE TYPE quest_routine AS ENUM ('daily', 'weekly', 'monthly', 'yearly', 'once');
+
+CREATE TYPE quest_action AS ENUM ('gm', 'vote', 'trade', 'gift', 'ticker', 'watchlist');
+
+CREATE TABLE IF NOT EXISTS quests (
+	id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+	title TEXT NOT NULL,
+	action quest_action NOT NULL,
+	frequency INT NOT NULL,
+	routine quest_routine NOT NULL DEFAULT 'daily'::quest_routine
+);
+
+INSERT INTO public.quests (title, action, frequency, routine) VALUES ('Say GM/GN', 'gm', 1, 'daily');
+INSERT INTO public.quests (title, action, frequency, routine) VALUES ('Vote for Mochi Bot on topgg/discordbotlist', 'vote', 2, 'daily');
+INSERT INTO public.quests (title, action, frequency, routine) VALUES ('Trading NFT', 'trade', 1, 'daily');
+INSERT INTO public.quests (title, action, frequency, routine) VALUES ('Gift XP for other people', 'gift', 1, 'daily');
+INSERT INTO public.quests (title, action, frequency, routine) VALUES ('Check any token price 3 times with $ticker', 'ticker', 3, 'daily');
+INSERT INTO public.quests (title, action, frequency, routine) VALUES ('Check your watchlist 3 times', 'watchlist', 3, 'daily');
+
+CREATE TABLE IF NOT EXISTS quests_reward_types (
+	id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+	name TEXT NOT NULL
+);
+
+INSERT INTO public.quests_reward_types (name) VALUES ('xp');
+INSERT INTO public.quests_reward_types (name) VALUES ('coin');
+
+CREATE TABLE IF NOT EXISTS quests_user_logs (
+	id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+	guild_id TEXT,
+	user_id TEXT NOT NULL,
+	quest_id UUID NOT NULL,
+	action quest_action NOT NULL,
+	target INT NOT NULL DEFAULT 0,
+	created_at timestamptz DEFAULT now(),
+	FOREIGN KEY (guild_id) REFERENCES discord_guilds(id)
+);
+
+CREATE TABLE IF NOT EXISTS quests_user_list (
+	id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+	user_id TEXT NOT NULL,
+	quest_id UUID NOT NULL,
+	action quest_action NOT NULL,
```

Comment:
maybe you miss default value for `action`

Reply:
this should be passed actively, so no default value imo